apps/logging/test_audit_loggers.py

```diff
@@ -0,0 +1,221 @@
+        self.client = Client()
+        # capture sls logging records
+        self.log_buffer_sls = io.StringIO()
+        self.channel_sls = logging.StreamHandler(self.log_buffer_sls)
+        self.channel_sls.setLevel(logging.INFO)
+        sls_logger.setLevel(logging.INFO)
+        sls_logger.addHandler(self.channel_sls)
+        # capture token logging records
+        self.log_buffer_token = io.StringIO()
+        self.channel_token = logging.StreamHandler(self.log_buffer_token)
+        self.channel_token.setLevel(logging.INFO)
+        token_logger.setLevel(logging.INFO)
+        token_logger.addHandler(self.channel_token)
+        # capture fhir data logging records
+        self.log_buffer_fhir = io.StringIO()
+        self.channel_fhir = logging.StreamHandler(self.log_buffer_fhir)
+        self.channel_fhir.setLevel(logging.INFO)
+        fhir_logger.setLevel(logging.INFO)
+        fhir_logger.addHandler(self.channel_fhir)
+
+    def tearDown(self):
+        # do not close stream, only close channel
+        # self.log_buffer_sls.close()
+        self.channel_sls.close()
+        # self.log_buffer_token.close()
+        self.channel_token.close()
+        # self.log_buffer_fhir.close()
+        self.channel_fhir.close()
+
+    def test_fhir_events_logging(self):
+        first_access_token = self.create_token('John', 'Smith')
+
+        @all_requests
+        def catchall(url, req):
+            fhir_log_content = self.log_buffer_fhir.getvalue()
+            self.assertIsNotNone(fhir_log_content)
+            log_entry_dict = json.loads(fhir_log_content)
+            self.assertEqual(log_entry_dict["fhir_id"], "-20140000008325")
+            self.assertEqual(log_entry_dict["user"], "patientId:-20140000008325")
+            self.assertEqual(log_entry_dict["path"], "/v1/fhir/Patient")
+            self.assertIsNotNone(log_entry_dict["application"])
+            return {
+                'status_code': 200,
+                # TODO replace this with true backend response, this has been post proccessed
+                'content': patient_response,
+            }
+
+        with HTTMock(catchall):
+            self.client.get(
+                reverse(
+                    'bb_oauth_fhir_search',
+                    kwargs={'resource_type': 'Patient'}),
+                {'count': 5, 'hello': 'world'},
+                Authorization="Bearer %s" % (first_access_token))
+            # check fhir log content
+            fhir_log_content = self.log_buffer_fhir.getvalue()
+            self.assertIsNotNone(fhir_log_content)
+            log_entries = fhir_log_content.splitlines()
+            log_entry_dict = json.loads(log_entries[0])
+            self.assertEqual(log_entry_dict["fhir_id"], "-20140000008325")
+            self.assertEqual(log_entry_dict["user"], "patientId:-20140000008325")
+            self.assertEqual(log_entry_dict["path"], "/v1/fhir/Patient")
+            self.assertIsNotNone(log_entry_dict["application"])
```

Comment:
@JFU-GIT There is also a `fhir_post_fetch` log event in the buffer. I tried out the following code to test it also. I really like this log capture and checking in the unit tests! TY for adding this to our testing methods!

```python
            log_entry_dict = json.loads(log_entries[0])
            self.assertEqual(log_entry_dict["type"], "fhir_pre_fetch")
            self.assertEqual(log_entry_dict["fhir_id"], "-20140000008325")
            self.assertEqual(log_entry_dict["user"], "patientId:-20140000008325")
            self.assertEqual(log_entry_dict["path"], "/v1/fhir/Patient")
            self.assertIsNotNone(log_entry_dict["application"])

            log_entry_dict = json.loads(log_entries[1])
            self.assertEqual(log_entry_dict["type"], "fhir_post_fetch")
            self.assertEqual(log_entry_dict["fhir_id"], "-20140000008325")
            self.assertEqual(log_entry_dict["user"], "patientId:-20140000008325")
            self.assertEqual(log_entry_dict["path"], "/v1/fhir/Patient")
            self.assertIsNotNone(log_entry_dict["application"])
```

Reply:
good catch, not able to fully check the other two entries!

Reply:
changes made in new push.